src/main/java/org/jabref/gui/groups/GroupSidePane.java

```diff
@@ -46,6 +89,21 @@ public Action getToggleAction() {
         return StandardActions.TOGGLE_GROUPS;
     }
 
+    private void toggleUnionIntersection() {
+        boolean intersectionToggled = preferences.getBoolean(JabRefPreferences.GROUP_INTERSECT_SELECTIONS);
+        boolean newState = !intersectionToggled;
+        preferences.putBoolean(JabRefPreferences.GROUP_INTERSECT_SELECTIONS, newState);
+
+        BasePanel basePanel = frame.getCurrentBasePanel();
```

Comment:
Please remove the dependency on `frame` and replace it by dialog service (which has a `notify` method).